extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/knative/events/CloudEventOutputBuilder.java

```diff
@@ -0,0 +1,75 @@
+package io.quarkus.funqy.knative.events;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+@SuppressWarnings({ "rawtypes", "unchecked" })
+public class CloudEventOutputBuilder {
+
+    public static class CloudEventOutput<T> {
```

Comment:
Pull CloudEventOutput up so its not nested.  The example I gave you was just so that I had one file to show you as semi-pseudocode.

Reply:
Would have to make your fields package protected though.

Reply:
extracted to new class file